### PR TITLE
Introduce SDK Configuration Properties for Creative Factory Timeout

### PR DESCRIPTION
**Pull request by jsligh**
Introduced:
- `PrebidMobile.creativeFactoryTimeout` to customize `BANNER_TIMEOUT`.
- `PrebidMobile.creativeFactoryTimeoutPreRenderContent` to customize `INTERSTITIAL_TIMEOUT` and `VAST_TIMEOUT`.
- In `bidrequest.ext.prebid.passthrough` used `cftbanner` for `BANNER_TIMEOUT` and `cftprerender` for `INTERSTITIAL_TIMEOUT` and `VAST_TIMEOUT`
- Added conditions in `CreativeFactoryTest` and `MobileSdkPassThroughTest`

### CODE REVIEW

File: PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/rendering/models/openrtb/bidRequests/MobileSdkPassThrough.java [hunk 1]
```diff
@@ -163,8 +168,19 @@ private MobileSdkPassThrough(JSONObject passThrough) {
...
             }
         } catch (JSONException exception) {
-            LogUtil.error(TAG, "Can't parse configuration");
+            LogUtil.error(TAG, "Can't parse adconfiguration");
+        }
+        try {
+            if (passThrough.has("sdkconfiguration")) {
+                configuration = passThrough.getJSONObject("sdkconfiguration");
+                getAndSave("cftbanner", Integer.class, it -> bannerTimeout = it);
+                getAndSave("cftprerender", Integer.class, it -> preRenderTimeout = it);
```
Comment (ValentinPostindustria) [inline PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/rendering/models/openrtb/bidRequests/MobileSdkPassThrough.java:177]: If a user set up only one configuration parameter this line will throw an exception. We should support cases with only one parameter.

File: PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/rendering/models/openrtb/bidRequests/MobileSdkPassThrough.java
Comment (ValentinPostindustria) [inline PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/rendering/models/openrtb/bidRequests/MobileSdkPassThrough.java:1]: For now `MobileSdkPassThrough` will be created only if it contains `adconfiguration`. But publishers can set up only `sdkconfiguration` parameters, without `adconfiguration`.

File: PrebidMobile/PrebidMobile-core/src/test/java/org/prebid/mobile/rendering/models/openrtb/bidRequests/MobileSdkPassThroughTest.java [hunk 2]
```diff
@@ -126,4 +126,20 @@ public void combine_checkFromBidPriority() throws JSONException {
         }
     }
 
+    @Test
+    public void create_putObjectWithSdkConfiguration_returnFullObject() throws JSONException {
+        if (!BuildConfig.DEBUG) {
+
+            JSONObject jsonObject = new JSONObject(
+                    "{\"prebid\":{\"passthrough\":[{\"type\":\"prebidmobilesdk\",\"adconfiguration\":{\n\"ismuted\": false,\n\"maxvideoduration\": 15,\n\"closebuttonarea\": 0.3,\n\"closebuttonposition\": \"topleft\",\n\"skipbuttonarea\": 0.3,\n\"skipbuttonposition\": \"topleft\",\n\"skipdelay\": 0}, \n\"sdkconfiguration\": {\n\"cftbanner\": 7800, \n\"cftprerender\": 21000}}]}}");
```
Comment (ValentinPostindustria) [inline PrebidMobile/PrebidMobile-core/src/test/java/org/prebid/mobile/rendering/models/openrtb/bidRequests/MobileSdkPassThroughTest.java:134]: Remove `adconfiguration`, leave `sdkconfiguration` and the test must pass.

File: PrebidMobile/PrebidMobile-core/src/test/java/org/prebid/mobile/rendering/models/openrtb/bidRequests/MobileSdkPassThroughTest.java [hunk 3]
```diff
@@ -126,4 +126,20 @@ public void combine_checkFromBidPriority() throws JSONException {
...
+
+            MobileSdkPassThrough subject = MobileSdkPassThrough.create(jsonObject);
+
+            assertNotNull(subject);
+            assertEquals((Integer) 7800, subject.bannerTimeout);
```
Comment (ValentinPostindustria) [inline PrebidMobile/PrebidMobile-core/src/test/java/org/prebid/mobile/rendering/models/openrtb/bidRequests/MobileSdkPassThroughTest.java:139]: Add separated tests with only bannerTimeout and only preRenderTimeout in Json.

**Review by ValentinPostindustria — CHANGES_REQUESTED** [review]

**Review by ValentinPostindustria — DISMISSED** [review]

**Comment by YuriyVelichkoPI** [comment]
Here is a task for adding the tests: https://github.com/prebid/prebid-mobile-autotests/issues/79

**Review by YuriyVelichkoPI — APPROVED** [review]
LGTM

**Pull request merged**